### ISSUE TITLE
Insert external node to folded parent will result with wrong position or error "No suitable position found to insert"

### ISSUE THREAD

**Issue by zushenyan** [fritz-c/react-sortable-tree#134]
# Category
bug.

# Environment
- Mac OS Sierra 10.12.5
- Chrome 59

### Note
I have similar issue on my local machine with the following modules' version:

- node@8.1.3

```
{
  "name": "react-dnd-test",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "prop-types": "^15.5.10",
    "react": "^15.6.1",
    "react-dnd": "^2.4.0",
    "react-dnd-html5-backend": "^2.4.1",
    "react-dom": "^15.6.1",
    "react-scripts": "1.0.10",
    "react-sortable-tree": "^0.1.21"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test --env=jsdom",
    "eject": "react-scripts eject"
  }
}
```

# Reproduce step
Go to [official example of dragging from external node](https://fritz-c.github.io/react-sortable-tree/storybook/index.html?selectedKind=Advanced&selectedStory=Drag%20from%20external%20source&full=0&down=1&left=1&panelRight=0&downPanel=storybook%2Fnotes%2Fpanel) page

### Scenario 1 (insert to the wrong position)

![insert_to_wrong_position](https://user-images.githubusercontent.com/829686/28871992-a8883a8e-77b9-11e7-9f9a-819e913b2bfb.gif)

1. Insert an external node to `node 1`.
2. Fold `node 1`.
3. Insert another external node to the `node 1`'s `external node` with auto expending.

Actual Result:
Will be inserted to `node 2`.

Expected Result:
Should be inserted to `node 1`'s `external node` as a child node.

### Scenario 2 (error "No suitable position found to insert")

![no_suitable_position](https://user-images.githubusercontent.com/829686/28872071-143b581a-77ba-11e7-8422-ff5de8cbe058.gif)

1. Insert an external node to `node 2`
2. Fold `node 2`.
3. Insert another external node to the `node 2`'s `external node` with auto expending.

Actual Result:
It throws "No suitable position found to insert".

Expected Result:
Insert node without errors.

**Comment by fritz-c** [comment]
Nice catch and very helpful explanation. I was just in the middle of working on another feature (tree-to-tree dragging) and this issue is fixed as a happy coincidence with that. I should be able to push out a stable version of that within a week. In the meantime, I've pushed the source to `master` if you want to clone and check it for yourself.

**Comment by fritz-c** [comment]
This should be solved in versions `>=1.0.0`.
